crates/common/mqtt_channel/src/session.rs

```diff
@@ -39,10 +40,45 @@ pub async fn init_session(config: &Config) -> Result<(), MqttError> {
                 break;
             }
 
-            Err(err) => {
-                eprintln!("Connection Error {}", err);
-                break;
-            }
+            Err(err) => match err {
+                rumqttc::ConnectionError::ConnectionRefused(ConnectReturnCode::BadClientId) => {
+                    return Err(MqttError::InitSessionError {
+                        reason: "bad client id".to_string(),
+                    });
+                }
+                rumqttc::ConnectionError::ConnectionRefused(
+                    ConnectReturnCode::BadUserNamePassword,
+                ) => {
+                    return Err(MqttError::InitSessionError {
+                        reason: "bad user name and password".to_string(),
+                    });
+                }
+                rumqttc::ConnectionError::ConnectionRefused(ConnectReturnCode::NotAuthorized) => {
+                    return Err(MqttError::InitSessionError {
+                        reason: " not authorized".to_string(),
+                    });
+                }
+                rumqttc::ConnectionError::ConnectionRefused(
+                    ConnectReturnCode::RefusedProtocolVersion,
+                ) => {
+                    return Err(MqttError::InitSessionError {
+                        reason: " protocol version mismatch".to_string(),
+                    });
+                }
+                rumqttc::ConnectionError::ConnectionRefused(
+                    ConnectReturnCode::ServiceUnavailable,
+                ) => {
+                    return Err(MqttError::InitSessionError {
+                        reason: " service not available".to_string(),
+                    });
+                }
+                rumqttc::ConnectionError::ConnectionRefused(ConnectReturnCode::Success) => {}
+                e => {
+                    return Err(MqttError::InitSessionError {
+                        reason: e.to_string(),
+                    });
+                }
+            },
```

Comment:
Nice. I would only move this code into an `impl From<rumqttc::ConnectionError> for MqttError>`.

Reply:
Refactored this code introducing a function to convert/map the errors.